### PR TITLE
Remaps the Lavaland Outpost. Again.

### PR DESCRIPTION
**Pull request by Sadhorizon**
## What Does This PR Do
This PR is meant to fix some issues with previous remap, as well as add a bunch of new features, that are meant to make it FEEL like an actual OUTPOST, instead of an elevator.

I'll go more over specific changes under "Images of Changes"

Also, on the code side of things, this reworks all of the outpost areas, adds code for an NT sign (we had the icon already), and adds a new pet.

## Why It's Good For The Game
Biggest issue of previous remap was shuttle touching the exit. Most of the outpost never saw the light of day.
By shuffling areas around, it will (hopefully) create more ways for people to go "Hey! Something is happening there! I might check it out! You know, halt gaming for a second and go talk with someone!"

Another big shuffle - medical being near the entrance. This just makes sense, you can stabalize injured faster.

And one huge addition - QM's office! Equipped with a bunch of consoles and the like. This PR also removes the handheld crew monitor from QM, in favor of a stationary one on lavaland.

Another new important feature - working disposals. Not much to say here. Disposaled items burn/get chasmed.

And lastly - working firelocks, including in the windows. A smart miner can press a switch and buy a few seconds against a fauna attack.

For less impactful additions, I added a LOT of fluff stuff. Coolest most definitely being Brad, the Mining Cockroach.

## Images of changes
Whole Remap (editor view)

**NOTE: I'll probably be too lazy to update images below. Map above is accurate at all times.**
**NOTE 2: I explain access required to get into a room below. "Mining Outpost" access is Mining Dock + Mining EVA.**
<details><summary>Mining Shuttle:</summary>
<p>
Mining shuttle is basically the same. I gave it a fire extinguisher, just in case, as well as a camera to easier check for intruders (aka, that drake that somehow got there)

You only need mining dock access to move to the outpost.

obraz

</p>
</details> 
<details><summary>Landing Pad:</summary>
<p>
Landing pad is a new addition - this is mostlly meant to be cool. Has docking beacons and a floodlight so you know where to land, a docking tube thingie so you can safely walk into the outpost. It also has a crate containing spare internals, just in case.

From here, you can get to other catwalks, west hallway, QM's Office, Life Support, TComms, EVA storage

obraz

obraz

</p>
</details> 
<details><summary>Other Catwalks</summary>
<p>
Catwalks are ALSO meant to be just a cool thing. QM can easily leave there, in case they need some fresh air. There is a loaded trashcart in the dark. There is also an abandoned crate, hiding just around a wall.
All of external airlocks need maintenance and mining outpost access types.

obraz

obraz

obraz

obraz

</p>
</details> 
<details><summary>QM's Office:</summary>
<p>
QM's office is a second brand new addition. QM can head over there to run away from responsibilities! Or, you know, handle mining... It contains a crew monitor console, a supply management console (for demoting miners on the spot), a guest pass terminal, a spare GPS, disposal bin, a newscaster and an empty locker (in case QM goes mining but wants to keep his gear safe). All the windows are also electrochromic, controlled by the button on QM's wall.
Airlocks are locked to QM's access, obviously.
From here, you can move either to west hall or outside.

obraz

obraz

</p>
</details> 
<details><summary>Locker Room:</summary>
<p>
It now has one less gear locker, but instead you can get some new clothes outpost-side.
To get inside, you need mining base access.

obraz

</p>
</details> 
<details><summary>West Hallway:</summary>
<p>
It contains a nice sitting area, mining shuttle console, a cigarette vendor, a nanomed, a disposal bin and Brad the Mining Cockroach. It also contains a notice board and a plaque reminding about the fallen.
From here you can get to the Mining Shuttle, Landing Pad, QM's Office, Locker Room, Life Support, South Maintenance, Cafeteria and East Hallway.
Internal Airlocks only need mining dock access.

obraz

obraz

obraz

</p>
</details> 
<details><summary>Life Support:</summary>
<p>
Greatly expanded from previous iteration. Inside, there is:
a) Power-Making Gear - SMES, PACMAN, power monitoring console
b) Atmospherics Gear - Air Tanks, pump shoving air into distro, atmospherics monitoring console, a gas flow meter, a portable pump, and a spare oxygen canister
c) Misc Gear - a cell charger, a hardhat (safety first!), a water tank, a pair of **budget** insuls

Doors need either the mining outpost access, or general engineering access (so that random engis can get in easier).
From this room, you can go to either external catwalk or west hallway.

obraz

</p>
</details> 
<details><summary>Tcomms:</summary>
<p>
Not much changed. I shuffled items that were on the walls and turned the airlock to be reinforced. It's a basic reinforcement, with metal and not plasteel.

obraz

</p>
</details> 
<details><summary>South Maintenance and Construction Site</summary>
<p>
This is THE maintenance area of new outpost.
Airlocks between rooms require mining outpost access and maintenance access. Internal airlocks require just maintenance access.
**Area Breakdown:**

obraz

obraz

This part of maints contains oil and welding fuel tanks, a box full of randomly-colored glowsticks, a rack with random tool and a crate with two vendor resupplies - cigarettes and nanomed (the wall one).
To the north from here there is west hall, to south is the production room, to north-east is mech bay, to south-east is more maints.

obraz

This part of maints contains a random arcade machine, a water tank and a locker. Inside of the locker, there is a random book, ten marker beacons and a maint loot spawner.
To the north-west, there is previous maint segment, to north-east there is a custodial closet, to south-east there is the construction site, and to south there is more maints.

obraz

This part of maints contains a random lighter and that's about it. To the north, there is previous maint segment. To the east, there is a catwalk.

obraz

This area is still under construction! Nanotrasen is yet unsure what should be here. It's been temporarily wired to south maintenance. There is a catwalk outside.
From here, you can get to either South Maintenance or East Hallway.

</p>
</details> 
<details><summary>Production Room:</summary>
<p>
To make it less suspicious to break inside here, there are two maint loot spawners inside now. Area is rather hard time-consuming to get to however, especially roundstart.
You may notice that the area is now connected to outpost directly. I don't think it's affecting sec/antag balance this much, as walls are pretty thick from all sides.

obraz

</p>
</details> 
<details><summary>East Hallway and Tool Storage:</summary>
<p>
Hallway contains a disposal bin, two mining vendors (as before), and a cigarette vendor. Tool Storage contains one electrical and two mechanical toolboxes. Disposal outlet is behind the tinted window.

Every airlock (except tool storage) here needs mining outpost access to be opened.

obraz

obraz

obraz

</p>
</details> 
<details><summary>Cafeteria:</summary>
<p>
Another big room rework - cafeteria now contains a counter, a freezer with donks and beer (amount is same as previously, just in the freezer), a grill, a spare top-hat (for the wannabe bartender), a newscaster, a disposals bin and the BEAUTIFUL view outside! There are also a couple vendors - snacks, drinks and coffee, as well as a grinder and a sink. When you are super bored, you can also watch the entertainment monitor.
Southern airlock needs mining dock access, while eastern one needs mining outpost access. Both lead to hallways.

obraz

obraz

</p>
</details> 
<details><summary>Mech Bay:</summary>
<p>
Mech bay contains two mech chargers, two borg chargers, a welding fuel tank and a machine repair kit. All you need to fix broken robots!
East airlock needs mining outpost access and leads to the east hallway, while the west airlock needs mining outpost and maintenance access, and leads to the southern maintenance.

obraz

</p>
</details> 
<details><summary>Custodial Storage:</summary>
<p>
Custodial Storage contains basic cleaning supplies - a mop, a mop bucket, a box of mixed light bulbs, a garbage bag and three "wet floor" signs.
East airlock needs mining outpost access and leads to the east hallway, while the western airlock needs mining outpost and maintenance access, and leads to the southern maintenance.

obraz

obraz

</p>
</details> 
<details><summary>Storage Room:</summary>
<p>
EVA Storage and Box Storage share the same area. Contents haven't changed, but the placement did.
All internal airlocks need mining outpost access.

obraz

Southern airlock leads to the east hallway, western airlock leads outside, eastern airlock leads to the box storage.

obraz

Southern doors lead to the outpost's airlock, western doors lead to the EVA storage.
</p>
</details> 
<details><summary>Infirmary:</summary>
<p>
This medbay is still pretty poorly equipped. Biggest change is it's new placement - near the front, where you need it the most.
It contains basic medical equipment - one medikit, one sleeper, one IV Drip, one blood bag of every type, some splints and one health analyzer upgrade.
Western doors need outpost access and lead to east hallway, northern doors need maint and outpost access and lead to the east maints.

obraz

obraz

</p>
</details> 
<details><summary>East Maintenance:</summary>
<p>
Contains the GPS console, a water tank and an emergency toolbox.
North doors lead to the outpost's airlock, south doors lead to the infirmary.

obraz

obraz

</p>
</details> 
<details><summary>Airlock:</summary>
<p>
Airlock Room is now bigger. It now contains a second shower and a nanomed.

obraz

</p>
</details> 
<details><summary>Demonstration of Fire Alarms:</summary>
<p>

obraz

obraz

</p>
</details> 
<details><summary>Cameras:</summary>
<p>

obraz

</p>
</details> 

## Testing
Spawned in, messed around with:
* disposals
* APCs
* firelocks
* access
* lighting
* a lot of lesser stuff

Nothing seemed to break.

## Changelog
:cl:
tweak: Remapped the mining outpost.
tweak: Renamed "Mining" access into "Mining Dock" access; Renamed "Mining EVA" access into "Mining Outpost" access.
tweak: Mining-related Quartermaster gear can now be found in outpost, instead of the station.
/:cl:

### CODE REVIEW

**Comment by Aylong220** [comment]
image
Unconnected scrubbers pipe

**Comment by Sadhorizon** [comment]
note for myself: 
wiki updates would be, adding mentions of second office to QM page, adding Brad to station pets page

**Comment by Oyu-De** [comment]
1. Is this cable here connected?
image

2. Why the S turn on the wiring? Just curious
image

**Comment by Sadhorizon** [comment]
Heyo! Thanks for the review!
> 1. Is this cable here connected?
>        image
Wire is there, yup!
>     2. Why the S turn on the wiring? Just curious
>        image
Tbh, it was leftover from an older design. I will just move the wire up!

**Review by S34NW — COMMENTED** [review]
image
You have some no lava helpers on walls

**Comment by Sadhorizon** [comment]
NO NO NO NO NO

**Review by S34NW — CHANGES_REQUESTED** [review]
Changes requested on discord, see from this message onwards https://discord.com/channels/145533722026967040/876666051297935410/1163941823849439374

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 1]
```diff
@@ -45,3 +45,11 @@
 /mob/living/simple_animal/cockroach/ex_act() //Explosions are a terrible way to handle a cockroach.
 	return
 
+//mining pet
+/mob/living/simple_animal/cockroach/Brad
+	name = "Brad"
+	desc = "Lavaland's most resilient cockroach. Seeing this little guy walk through the wastes almost makes you wish for nuclear winter."
+	response_help  = "carefully pets"
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm:52]:
```suggestion
	response_help = "carefully pets"
```

File: _maps/map_files/generic/Lavaland.dmm
Comment (S34NW) [inline _maps/map_files/generic/Lavaland.dmm:1]: image

window spawners and buttons have ID vars, this needs them too

**Review by S34NW — CHANGES_REQUESTED** [review]

File: _maps/map_files/generic/Lavaland.dmm
Comment (S34NW) [reply _maps/map_files/generic/Lavaland.dmm:1]: PR is fine apart from this

**Review by S34NW — APPROVED** [review]

**Pull request merged**